src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -316,6 +308,78 @@ public LockFileBuilder(int lockFileVersion, ILogger logger, Dictionary<RestoreTa
             return lockFile;
         }
 
+        private static void AddProjectFileDependenciesForSpec(PackageSpec project, LockFile lockFile)
+        {
+            // Use empty string as the key of dependencies shared by all frameworks
+            lockFile.ProjectFileDependencyGroups.Add(new ProjectFileDependencyGroup(
+                string.Empty,
+                project.Dependencies
+                    .Select(group => group.LibraryRange.ToLockFileDependencyGroupString())
+                    .OrderBy(group => group, StringComparer.Ordinal)));
```

Comment:
Ordering in the lock file is done purely to produce the same result each time, and Ordinal is faster than OrdinalIgnoreCase